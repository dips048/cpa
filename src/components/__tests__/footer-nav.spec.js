import React from "react"
import renderer from "react-test-renderer"
import FooterNav from "../footer-nav"
import { render } from "@testing-library/react"

describe("FooterNav", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<FooterNav companyName={"MyCompany"} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test("should render the default list (contact) with `companyName", () => {
    const { getByText } = render(<FooterNav companyName={"MyCompany"} />)
    expect(getByText("Contact")).toBeInTheDocument()
    expect(getByText("support@cpasitesolutions.com")).toBeInTheDocument()
    expect(getByText("© MyCompany 2022")).toBeInTheDocument()
    expect(getByText("Site Map")).toBeInTheDocument()
    expect(getByText("Privacy Policy")).toBeInTheDocument()
    expect(getByText("Disclaimer")).toBeInTheDocument()
  })

  test("when `menuLinks` contains the Resources links it should render the `resources` links", () => {
    const menuLinks = [
      {
        name: "Resources",
        url: "/resources",
        subLinks: [
          {
            name: "Newsletter",
            url: "/newsletter",
            subLinks: [],
          },
          {
            name: "Guides",
            url: "/guides",
            subLinks: [],
          },
        ],
      },
      {
        name: "Services",
        url: "/services",
        subLinks: []
      }
    ]

    const { getByText } = render(
      <FooterNav companyName={"MyCompany"} menuLinks={menuLinks} />
    )
    expect(getByText("Resources")).toBeInTheDocument()
    expect(getByText("Newsletter")).toBeInTheDocument()
    expect(getByText("Guides")).toBeInTheDocument()
    expect(getByText("Contact")).toBeInTheDocument()
    expect(getByText("support@cpasitesolutions.com")).toBeInTheDocument()
    expect(getByText("© MyCompany 2022")).toBeInTheDocument()
    expect(getByText("Site Map")).toBeInTheDocument()
    expect(getByText("Privacy Policy")).toBeInTheDocument()
    expect(getByText("Disclaimer")).toBeInTheDocument()
  })

  test("when `menuLinks` contains the Tax Center links it should render the `Tax Center` links", () => {
    const menuLinks = [
      {
        name: "Tax Center",
        url: "/tax-center",
        subLinks: [
          {
            name: "Track Your Refund",
            url: "/track-your-refund",
            subLinks: [],
          },
          {
            name: "Tax Due Dates",
            url: "/tax-due-dates",
            subLinks: [],
          },
        ],
      },
    ]

    const { getByText } = render(
      <FooterNav companyName={"MyCompany"} menuLinks={menuLinks} />
    )
    expect(getByText("Tax Center")).toBeInTheDocument()
    expect(getByText("Track Your Refund")).toBeInTheDocument()
    expect(getByText("Tax Due Dates")).toBeInTheDocument()
    expect(getByText("Contact")).toBeInTheDocument()
    expect(getByText("support@cpasitesolutions.com")).toBeInTheDocument()
    expect(getByText("© MyCompany 2022")).toBeInTheDocument()
    expect(getByText("Site Map")).toBeInTheDocument()
    expect(getByText("Privacy Policy")).toBeInTheDocument()
    expect(getByText("Disclaimer")).toBeInTheDocument()
  })
})
