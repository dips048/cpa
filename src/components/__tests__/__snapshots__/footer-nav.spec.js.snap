// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterNav renders correctly 1`] = `
<div
  className="footer-nav"
>
  <div
    className="container"
  >
    <div
      className="linkConatainer"
    >
      <ul
        className="navLinks"
      >
        <li>
          <h3>
            Contact
          </h3>
        </li>
        <li>
          MyCompany
        </li>
        <li>
          <a
            href="tel:(800) 896-4500"
          >
            (800) 896-4500
          </a>
        </li>
        <li>
          <a
            href="mailto:support@cpasitesolutions.com"
          >
            support@cpasitesolutions.com
          </a>
        </li>
      </ul>
    </div>
    <hr />
    <div
      className="linkConatainer"
    >
      <ul
        className="linkConatainerB"
      >
        <li
          className="ps-2"
        >
          <span>
            Â© 
            MyCompany
             2022
          </span>
        </li>
        <li
          className="ps-2"
        >
          <a
            href="?"
          >
            Site Map
          </a>
        </li>
        <li
          className="ps-2"
        >
          <a
            href="?"
          >
            Privacy Policy
          </a>
        </li>
        <li
          className="ps-2"
        >
          <a
            href="?"
          >
            Disclaimer
          </a>
        </li>
      </ul>
      <div
        className="text-center"
      >
        <a
          className="m-1"
          href="https://www.facebook.com"
          title="Facebook"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-facebook "
            data-icon="facebook"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.8 90.69 226.4 209.3 245V327.7h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.3 482.4 504 379.8 504 256z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
        <a
          className="m-1"
          href="https://www.twitter.com"
          title="Twitter"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-twitter "
            data-icon="twitter"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.4 151.7c.325 4.548 .325 9.097 .325 13.65 0 138.7-105.6 298.6-298.6 298.6-59.45 0-114.7-17.22-161.1-47.11 8.447 .974 16.57 1.299 25.34 1.299 49.06 0 94.21-16.57 130.3-44.83-46.13-.975-84.79-31.19-98.11-72.77 6.498 .974 12.99 1.624 19.82 1.624 9.421 0 18.84-1.3 27.61-3.573-48.08-9.747-84.14-51.98-84.14-102.1v-1.299c13.97 7.797 30.21 12.67 47.43 13.32-28.26-18.84-46.78-51.01-46.78-87.39 0-19.49 5.197-37.36 14.29-52.95 51.65 63.67 129.3 105.3 216.4 109.8-1.624-7.797-2.599-15.92-2.599-24.04 0-57.83 46.78-104.9 104.9-104.9 30.21 0 57.5 12.67 76.67 33.14 23.72-4.548 46.46-13.32 66.6-25.34-7.798 24.37-24.37 44.83-46.13 57.83 21.12-2.273 41.58-8.122 60.43-16.24-14.29 20.79-32.16 39.31-52.63 54.25z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
        <a
          className="m-1"
          href="https://www.linkedin.com"
          title="LinkedIn"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-linkedin "
            data-icon="linkedin"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</div>
`;
